### ISSUE TITLE
browser did not display wallet balance breakdown correctly

### ISSUE THREAD

**Issue by LaurenWags** [brave/browser-laptop#14693]
### Description
Reported via community, a user saw that 30 BAT was taken from their wallet for a 15 BAT contribution. (https://community.brave.com/t/funds-not-credited/25528/11)

Further inspection of the txn history showed that the correct amount was removed, so it seems as though the browser did not display the wallet breakdown as expected.

### Steps to Reproduce
Unknown exactly, here's a summary from the user:
Have some BAT in wallet and 15 bonus BAT from Brave:

![rewer1](https://user-images.githubusercontent.com/28145373/42452420-f013664e-8357-11e8-9275-1b1569ac026c.png)

My monthly budget was set to 15 BAT, and the latest payment was processed on 6/26.  

![rewer2](https://user-images.githubusercontent.com/28145373/42452456-0708189a-8358-11e8-9663-39aacb5219ed.png)

Contribution statement shows a contribution of 15 BAT:
![rewer3](https://user-images.githubusercontent.com/28145373/42452510-340d0ef4-8358-11e8-8023-dbb475123560.png)

**Actual result:**
looks like 30 BAT was taken from user's wallet because balance was not displaying properly

**Expected result:**
browser should display wallet balance correctly

**Reproduces how often:**
Unknown

### Brave Version

**about:brave info:**
0.22.810

**Reproducible on current live release:**
unknown

### Additional Information
I have the paymentId for investigation. Please contact me for it.

**Comment by LaurenWags** [comment]
Further inspection of the txn history for this particular case from @evq showed that:
1. 02/24 UGP -> 10 BAT -> card
2. 02/24 card -> 5 BAT -> settlement
3. 03/20 BTC in -> 8 BAT -> card
4. 03/26 card -> 5 BAT -> settlement
5. 05/13 card -> 5 BAT -> settlement
(balance = 3 BAT)
6. 06/19 BTC in -> 37.6 BAT -> card
(balance = 40.6 BAT)
7. 06/20 UGP -> 15 BAT -> card
8. 06/26 card -> 15 BAT -> settlement
(balance = 40.6 BAT)

Seems as though the browser wasn't displaying the balance correctly

**Comment by elsaber** [comment]
Hi @LaurenWags,

This is me at GitHub :)

If any questions arise, feel free to ask.

**Comment by BrendanEich** [comment]
I observed what this issue reports, but only in between when I think reconciliation started (a day after the notification infobar) and when it completed. I saw this after reconciliation starting:

![image](https://user-images.githubusercontent.com/313317/43226048-93fa548e-900f-11e8-9a39-eca02cb8afe8.png)

and then a day later, all was well:

![image](https://user-images.githubusercontent.com/313317/43226081-a8c94a32-900f-11e8-9bd9-5cc29e4fe4f9.png)

Sorry if this is a distinct bug from the one reported by this issue. It seems purely a client-side bug where we mis-represent the state of things. @evq  kindly confirmed via my privately-communicated paymentId that my balance after reconciliation is correct.
